module-system/sov-modules-stf-template/src/lib.rs

```diff
@@ -97,33 +115,42 @@ where
 
     type Witness = <<C as Spec>::Storage as Storage>::Witness;
 
-    fn init_chain(&mut self, params: Self::InitialState) {
+    type Condition = Cond;
+
+    fn init_chain(&mut self, params: Self::InitialState) -> anyhow::Result<jmt::RootHash> {
         let mut working_set = StateCheckpoint::new(self.current_storage.clone()).to_revertable();
 
         self.runtime
             .genesis(&params, &mut working_set)
             .expect("module initialization must succeed");
 
         let (log, witness) = working_set.checkpoint().freeze();
-        self.current_storage
+        let genesis_hash = self
+            .current_storage
             .validate_and_commit(log, &witness)
             .expect("Storage update must succeed");
+
+        Ok(jmt::RootHash(genesis_hash))
     }
 
-    fn apply_slot<'a, I>(
+    fn apply_slot<'a, I, Data>(
         &mut self,
         witness: Self::Witness,
+        slot_data: &Data,
         blobs: I,
-    ) -> SlotResult<
-        Self::StateRoot,
-        Self::BatchReceiptContents,
-        Self::TxReceiptContents,
-        Self::Witness,
+    ) -> anyhow::Result<
```

Comment:
Just a reminder that this `anyhow::Result` is not needed

Reply:
Fixed in the last PR